lightning/src/events/mod.rs

```diff
@@ -1541,11 +1570,18 @@ impl Writeable for Event {
 					(9, channel_funding_txo, option),
 				});
 			},
-			&Event::DiscardFunding { ref channel_id, ref transaction } => {
+			&Event::DiscardFunding { ref channel_id, ref funding_info } => {
 				11u8.write(writer)?;
+
+				let transaction = if let FundingInfo::Tx { transaction } = funding_info {
+					Some(transaction)
+				} else {
+					None
+				};
 				write_tlv_fields!(writer, {
 					(0, channel_id, required),
-					(2, transaction, required)
+					(2, transaction, option),
+					(4, funding_info, required),
```

Comment:
Tried to look into ways to not break downgrades for this (i.e. writing a dummy or partially-dummy tx if `unsafe_manual_funding_transaction_generated` was used) but it doesn't seem worth it. So just noting that we'll need a release note for this since I don't see one in `pending_changelog`.

Reply:
I can ask how the `pending_changelog` works? I could make the followup, sorry if I missed it